services/management/backup/locations_service.go

```diff
@@ -215,16 +257,46 @@ func convertLocation(location *models.BackupLocation) (*backupv1beta1.Location,
 	return loc, nil
 }
 
-// RemoveLocation removes backup location.
-func (s *LocationsService) RemoveLocation(ctx context.Context, req *backupv1beta1.RemoveLocationRequest) (*backupv1beta1.RemoveLocationResponse, error) {
-	mode := models.RemoveRestrict
-	if req.Force {
-		mode = models.RemoveCascade
+func (s *LocationsService) getBucketLocation(c *models.S3LocationConfig) (string, error) {
+	url, err := models.ParseEndpoint(c.Endpoint)
+	if err != nil {
+		return "", status.Errorf(codes.InvalidArgument, "%s", err)
 	}
-	if err := models.RemoveBackupLocation(s.db.Querier, req.LocationId, mode); err != nil {
-		return nil, err
+
+	secure := true
+	if url.Scheme == "https" {
+		secure = false
 	}
-	return &backupv1beta1.RemoveLocationResponse{}, nil
+
+	bucketLocation, err := s.s3.GetBucketLocation(url.Host, secure, c.AccessKey, c.SecretKey, c.BucketName)
+	if err != nil {
+		return "", err
+	}
+
+	return bucketLocation, nil
+}
+
+func (s *LocationsService) checkBucket(c *models.S3LocationConfig) error {
+	url, err := models.ParseEndpoint(c.Endpoint)
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "%s", err)
+	}
+
+	secure := true
+	if url.Scheme == "https" {
+		secure = false
+	}
+
+	exists, err := s.s3.BucketExists(url.Host, secure, c.AccessKey, c.SecretKey, c.BucketName)
+	if err != nil {
+		return err
```

Comment:
The user will see `Internal error` if credentials not valid. And, it looks like, without any descriptive message from s3 provider.